shared/js/ui/base/model.es6.js

```diff
@@ -123,17 +129,30 @@ BaseModel.prototype = $.extend({},
         },
 
 
-         /**
-          * Private method for turning `this` into a
-          * JSON object before sending to application store.
-          * Basically just weeds out properties that
-          * are functions.
-          */
-         _toJSON: function () {
-             let attributes = Object.assign({}, Object.getPrototypeOf(this), this);
-             if (attributes.store) delete attributes.store;
-             return JSON.parse(JSON.stringify(attributes));
-         }
+        /**
+         * Private method for turning `this` into a
+         * JSON object before sending to application store.
+         * Basically just weeds out properties that
+         * are functions.
+         */
+        _toJSON: function () {
+            let attributes = Object.assign({}, Object.getPrototypeOf(this), this);
+            if (attributes.store) delete attributes.store;
+            return JSON.parse(JSON.stringify(attributes));
+        },
+
+        /**
+         * Generate unique identifier
+         * (from http://stackoverflow.com/a/2117523)
+         */
+        _getUUID: function () {
+            return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function (c) {
+                let r = Math.random() * 16 | 0
+                let v = c == 'x' ? r : (r & 0x3 | 0x8)
```

Comment:
This will likely give you some trouble when you add linting. May have to add some ignores.